tests/test_plugin_loader.py

```
@@ -2310,6 +2310,403 @@ def test_clone_or_update_repo_checkout_fails_fallback(
+    def test_clone_or_update_repo_validated_handles_update_exception(
+        self, _mock_isdir, mock_update_commit, mock_logger
+    ):
+        """Validated update path should catch and log raised git errors."""
+        mock_update_commit.side_effect = subprocess.TimeoutExpired("git", 120)
+
+        result = pl._clone_or_update_repo_validated(
+            "https://github.com/user/repo.git",
+            "commit",
+            "deadbeef",
+            "repo",
+            "/tmp/plugins",
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/plugins"

(S108)